SwahiliBox/swahilibox-events-api
# WIP: Refactor and add validation tests

#### What does this PR do?
- Refactors code for creating a user
- Adds response wrapper for responding to users
- Adds end to end testing for all validation failure cases

#### Description of Task to be completed?


#### How should this be manually tested?
- Start postgres db and run on terminal `npm start`
- Start postman and set to `POST` `localhost:2000/signup`
- Set up a user object with properties `firstName`, `lastName`, `password`, `email`
- Send and you should get a successful response with the user object

#### Any background context you want to provide?
None

#### What are the relevant tasks or issues?
https://github.com/SwahiliBox/swahilibox-events-api/issues/32

#### Screenshots (if appropriate)
None

#### Questions:
Can someone please comment on how to make it more towards oop since I'm not much of an oop person
